osbs/build/build_request.py

```diff
@@ -1186,7 +1195,22 @@ def render(self, validate=True):
         if koji_task_id is not None:
             self.set_label('koji-task-id', str(koji_task_id))
 
-        self.template['spec']['nodeSelector'] = self.platform_node_selector
+        # for worker builds set nodeselectors
+        if self.spec.platforms.value is None:
```

Comment:
I'd rather this was a check for `platform`, not `platforms`. There's also a `build_type` spec value now, too. It's probably better to use `build_type` because we really are checking the build type here. Checking `platform` or `platforms` is just a heuristic for achieving that.